Rewrite multicluster installation docs (#8180)
istio/istio.io: content/en/docs/setup/install/multicluster/_index.md

> ---

Reviewer: Rename this file: `_index.md` -> `index.md`

keywords: [kubernetes,multicluster]

> test: n/a

Reviewer: ```suggestion
test: yes
owner: istio/wg-environments-maintainers
```

---
When installing Istio across multiple clusters, there are more things to
consider than with a [single cluster installation](/docs/setup/getting-started).

> There is no default installation procedure for a multicluster mesh, as it will
> be different based on your cluster and network architecture. We therefore
> provide options for:

Reviewer: ```suggestion
Although there is no default installation procedure for a multicluster mesh, as it will
be different based on your cluster and network architecture, this guide covers some
of the most common setups:
provide options for:
```

- [Network topologies](/docs/ops/deployment/deployment-models#network-models):
  one or two networks

- [Control plane topologies](/docs/ops/deployment/deployment-models#control-plane-models):
  multi-primary, primary-remote

> Before you begin, review the [deployment models](/docs/ops/deployment/deployment-models)
> which provides foundational concepts used throughout this guide.

Reviewer: ```suggestion
Before you begin, review the [deployment models guide](/docs/ops/deployment/deployment-models)
which describes the foundational concepts used throughout this guide.
```

## Requirements

This guide requires that you have two Kubernetes clusters with any of the
[supported versions](/docs/setup/platform-setup).

If the clusters are on different networks (i.e. no direct connectivity between
pods on different clusters), they must have ingress gateway services which are
accessible from the other cluster, ideally using L4 network load balancers
(NLBs). Not all cloud providers support NLBs and some require special
annotations to use them, so please consult your cloud provider’s documentation
for enabling NLBs for service object type load balancers. When deploying on
platforms without NLB support, it may be necessary to modify the health checks
for the load balancer to register the ingress gateway.

## Environment Variables

> This guide will refer to two clusters named `FRED` and `BARNEY`. The following

Reviewer: From the last doc WG call, I had the TODO to find out if the Flintstones reference in this document is [too culturally specific](https://developers.google.com/style/translation#be-inclusive).

I think the answer is that's fine. From IBM doc team: `unless the knowledge of the reference is required to understand what's happening in the example, then it doesn't really matter ... you dont need to know anything about the flintstones to understand the interactions between the clusters or anything, right? if not i think it's cool to keep it as-is. doesnt really matter if it is lost on people bc the reference isnt required to understand what's happening`

Reviewer 2: That linked doc does go on to say:

`Don't use the names of fictional characters from material that's in copyright. (This includes virtually all movies and TV shows.)`

While I think we would be OK (and indeed while I suggested we do something like this), I agree that changing it would be sensible.

Bike-shedding on names now open on Slack :)

Reviewer: Actually, the IBMer I talked to did also say `the only other thing to consider is copyright infringement but doesnt look like there's any of that here, just general names`. Basically, she thinks it's OK because we're not actually using the name `Flintstones`, just `Fred` and `Barney`. Anyway, if there's any doubt I agree we should change it.

Author: @craigbox any thoughts? Are you fine going with the names as-is or do you have a suggestion for a change?

Reviewer 2: Hmm.

Your networks were named WILMA and BETTY and I renamed them EAST and WEST because people who know the Flintstones would have thought they were the same class of object.

Now that's causing trouble with east-west traffic I see..

I checked in with a TW, and her guidance was "use CLUSTER-1, CLUSTER-2, NETWORK-1, NETWORK-2."  That's in line with Google docs guidance, any opinion from IBM?

I'd say "go with what we have" and look at new guidelines and updates later in a separate commit.
